### PR TITLE
upgraded Qt to 5.12.11 without cherry-picking - bitcoin fixes this (by copy-paste)

### PR DESCRIPTION
**Pull request by fmhc**
Can build depends & core with upgraded Qt 5.12.11 on Ubuntu 20.04.
Instead of fine selected git cherry-picks which would involve switching to a newer build, i just got the new stuff from the bitcoins in and got this flight-ready... My tests:
image

### CODE REVIEW

**Comment by fmhc** [comment]
https://github.com/dogecoin/dogecoin/pull/2339 this was the PR before, had to start over - multiple times and in the end gave up cherry-picking through 60+ commits and keeping out a change of the build system i guess.

**Comment by rnicoll** [comment]
Thanks for this! I do want to get this to build on Gitian before we merge, and currently I get the following error on Gitian:

```
ERROR: Feature 'c++1z' was enabled, but the pre-condition 'features.c++14 && tests.c++1z' failed.
make: *** [/home/ubuntu/build/dogecoin/depends/work/build/x86_64-apple-darwin11/qt/5.12.11-4e56e558921/./.stamp_configured] Error 3
make: Leaving directory `/home/ubuntu/build/dogecoin/depends'
```

If you have ideas, please do take a look, or I can take a look at this next week.

File: depends/packages/qt.mk [hunk 1]
```diff
@@ -1,55 +1,56 @@
...
 $(package)_config_opts += -no-iconv
 $(package)_config_opts += -no-kms
 $(package)_config_opts += -no-linuxfb
+$(package)_config_opts += -no-libjpeg
+$(package)_config_opts += -no-libproxy
 $(package)_config_opts += -no-libudev
-$(package)_config_opts += -no-mitshm
 $(package)_config_opts += -no-mtdev
-$(package)_config_opts += -no-pulseaudio
+$(package)_config_opts += -no-openssl
```
Comment (fdoving) [inline depends/packages/qt.mk:49]: Have a closer look at this.

Comment (fmhc) [reply depends/packages/qt.mk:49]: This is for the Qt build, but I am unsure for what this is used. My guess (unprooved!!!) was that Qt is communicating without SSL to the local socket where dogecoind is running and openssl for the network communication is added at some other package. But I'm new to Qt, really, and I just copied this dependency upgrade from bitcoin and it was working for me. I am currently setting up a gitian build and will dive into these things a bit deeper, also test building for Windows.

**Comment by patricklodder** [comment]
removed from 1.14.4 target milestone, qt (and openssl) dependency updates will have to target 1.14.5

File: depends/packages/qt.mk (hunk 1 above)
Comment (ReverseControl) [reply depends/packages/qt.mk:49]: @fmhc when in doubt never remove OpenSSL from anything. You need a reason to remove it.

**Comment by rnicoll** [comment]
I think we can postpone to 1.14.6 or leave for 1.21 to resolve, presuming #2595

**Review by dgs2019 — APPROVED** [review]

**Comment by rnicoll** [comment]
Rebased as https://github.com/rnicoll/dogecoin/tree/issue-2339-upgrade-qt-to-5.12.11-nocherrypicking - will see if it passes CI this time, but that's a blocker for reviewing

**Comment by patricklodder** [comment]
I think this won't work - no OpenSSL. This should be doable now with openssl 1.0.2u - I have some code just haven't touched it yet (I'm currently working on https://github.com/dogecoin/dogecoin/discussions/2624#discussioncomment-1509897 and just started SSE2 work, so this would be something I can pick up next week)

**Comment by patricklodder** [comment]
Closing this, like #2339, Qt 5.15 is unmaintained - we'd be wise to integrate 6.x if we can.

**Pull request closed without merging**